src/main/java/org/openrewrite/java/migrate/lang/UseVarKeyword.java

```diff
@@ -0,0 +1,162 @@
+import org.openrewrite.java.JavaVisitor;
+import org.openrewrite.java.search.HasJavaVersion;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+import org.openrewrite.java.tree.TypeTree;
+
+import java.time.Duration;
+
+import static java.lang.String.format;
+import static java.util.Objects.*;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class UseVarKeyword extends Recipe {
+    public String getDisplayName() {
+        return "Use local variable type-inference (var) where possible";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Local variable type-inference reduce the noise produces by repeating the type definitions in Java 10 or higher.";
+    }
+
+    @Override
+    public @Nullable Duration getEstimatedEffortPerOccurrence() {
+        return Duration.ofMinutes(1);
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getSingleSourceApplicableTest() {
+        return new HasJavaVersion("10", true).getVisitor();
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getVisitor() {
+        // J.VariableDeclarations
+        return new JavaVisitor<ExecutionContext>() {
+            private final JavaType.Primitive SHORT_TYPE = JavaType.Primitive.Short;
+            private final JavaType.Primitive BYTE_TYPE = JavaType.Primitive.Byte;
+            private final JavaTemplate template = JavaTemplate.builder(this::getCursor, "var #{} = #{any()}")
+                    .javaParser(JavaParser.fromJavaVersion()).build();
+
+            @Override
+            public J.VariableDeclarations visitVariableDeclarations(J.VariableDeclarations multiVariable, ExecutionContext executionContext) {
+                J.VariableDeclarations vd = (J.VariableDeclarations) super.visitVariableDeclarations(multiVariable, executionContext);
+
+                boolean isOutsideMethode = !determineIfIsInsideMethode(this.getCursor());
+                boolean isMethodeParameter = determineIfMethodeParameter(vd, this.getCursor());
+                if (isOutsideMethode || isMethodeParameter) return vd;
+
+                TypeTree typeExpression = vd.getTypeExpression();
+                boolean isByteVariable = typeExpression instanceof J.Primitive && BYTE_TYPE.equals(typeExpression.getType());
+                boolean isShortVariable = typeExpression instanceof J.Primitive && SHORT_TYPE.equals(typeExpression.getType());
+                if (isByteVariable || isShortVariable) return vd;
+
+                boolean definesNoVariable = vd.getVariables().isEmpty();
+                boolean isCompoundDefinition = vd.getVariables().size() < 1;
+                boolean isPureAssigment = isNull(vd.getTypeExpression());
+                if (definesNoVariable || isCompoundDefinition || isPureAssigment) return vd;
+
+                Expression initializer = vd.getVariables().get(0).getInitializer();
```

Comment:
The initializer should be unwrapped, if it is inside parentheses, otherwise you end up with `var` when you have `Object o = (null);`, which wouldn't compile. You can call `Expression#unwrap()` for this. But even better is probably if you call `getType()` on the expression and check against `JavaType.Primitive.Null`.

Reply:
Oh yes! I searched for `Primitive.Null` but somehow without success. Cheers